code/campaignclassic/src/main/java/com/adobe/marketing/mobile/CampaignPushTrackerActivity.java

```diff
@@ -82,22 +82,6 @@ private void handlePushOpen(final Intent intent) {
      */
     private void handlePushButtonClicked(final Intent intent) {
         CampaignClassic.trackNotificationClick(getTrackInfo(intent));
-
-        // Dismiss the notification once interacted
-        final String messageId =
-                intent.getStringExtra(CampaignPushConstants.Tracking.Keys.MESSAGE_ID);
-        if (!StringUtils.isNullOrEmpty(messageId)) {
-            NotificationManager notificationManager =
-                    (NotificationManager)
```

Comment:
I recollect that we use [`tag` or `messageId` for constructing the Notification identifier](https://github.com/adobe/aepsdk-campaignclassic-android/blob/06d4150da794b02ff00aaa96a9b80a41745b5e76/code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPMessagingService.java#L58). Have we checked if the issue happens only for notifications with missing `messageId` but have a `tag` instead ( because we check messageId for null/empty but not tag).

Reply:
when we assign the tag value we check the push template payload first to see if its present. if it isn't, we take the message id instead and assign it as the tag. This removed code wasn't working because I made a change to schedule notifications with the notification manager using the tag rather than the message id. This code would work if no tag was supplied in the push payload which would then mean the tag and message id would be the same.

in addition, the auto cancel setting was only applied to the notification itself. The new code to handle the dismissal of the notification on action pressed is the proper way to do it.

Reply:
Got it. Thank you for clarifying.